src/main/java/org/craftercms/studio/impl/v2/upgrade/operations/RenameUpgradeOperation.java

```diff
@@ -24,48 +24,54 @@
 import java.io.File;
 import java.nio.file.Path;
 import java.nio.file.Paths;
-import static org.craftercms.studio.api.v1.util.StudioConfiguration.BLUE_PRINTS_PATH;
 
 /**
  * Implementation of {@link org.craftercms.studio.api.v2.upgrade.UpgradeOperation} that renames the blueprint in the
  * global repository.
  * @author Dejan Brkic
  */
-public class BlueprintRenameFolderUpgradeOperation extends AbstractUpgradeOperation {
+public class RenameUpgradeOperation extends AbstractUpgradeOperation {
 
-    public static final String CONFIG_KEY_BLUEPRINT_FOLDER_NAME = "blueprintFolderName";
-    public static final String CONFIG_KEY_NEW_FOLDER_NAME = "newFolderName";
+    public static final String CONFIG_KEY_SITE = "site";
+    public static final String CONFIG_KEY_OLD_PATH = "oldPath";
+    public static final String CONFIG_KEY_NEW_PATH = "newPath";
 
-    protected String blueprintFolderName;
-    protected String newFolderName;
+    protected String siteId;
+    protected String oldPath;
+    protected String newPath;
 
     /**
      * {@inheritDoc}
      */
     @Override
     public void doInit(final Configuration config) {
-        blueprintFolderName = config.getString(CONFIG_KEY_BLUEPRINT_FOLDER_NAME);
-        newFolderName = config.getString(CONFIG_KEY_NEW_FOLDER_NAME);
+        siteId = config.getString(CONFIG_KEY_SITE);
+        oldPath = config.getString(CONFIG_KEY_OLD_PATH);
+        newPath = config.getString(CONFIG_KEY_NEW_PATH);
     }
 
     @Override
     public void execute(final String site) throws UpgradeException {
         try {
-            Path globalConfigPath = getRepositoryPath(StringUtils.EMPTY);
-            Path blueprintPath = Paths.get(globalConfigPath.toAbsolutePath().toString(),
-                    studioConfiguration.getProperty(BLUE_PRINTS_PATH), blueprintFolderName);
-            File blueprintFolder = blueprintPath.toFile();
-            if (blueprintFolder.exists()) {
-                Path newBlueprintPath = Paths.get(globalConfigPath.toAbsolutePath().toString(),
-                        studioConfiguration.getProperty(BLUE_PRINTS_PATH), newFolderName);
-                File newBlueprintFolder = newBlueprintPath.toFile();
-                FileUtils.deleteDirectory(newBlueprintFolder);
-                blueprintFolder.renameTo(newBlueprintFolder);
+
+            Path oldP = Paths.get(getRepositoryPath(siteId).getParent().toAbsolutePath().toString(), oldPath);
+            Path newP = Paths.get(getRepositoryPath(siteId).getParent().toAbsolutePath().toString(), newPath);
+
+            File oldF = oldP.toFile();
+            File newF = newP.toFile();
+            if (newF.exists()) {
+                FileUtils.forceDelete(newF);
```

Comment:
If the new path already exists it means the rename was done either by someone manually or by other upgrade, I think instead of deleting it (very dangerous) the class should end without doing any changes

Reply:
Currently, without deleting, it will never execute because 'BlueprintsUpgradeOperation` runs first and copies renamed blueprints.

Reply:
ok then it would be better to add a property `overwrite` that defaults to false, and we set it to true for this one in the yaml